src/main/java/uk/gov/justice/probation/courtcasematcher/messaging/HearingExtractor.java

```diff
@@ -63,11 +71,26 @@ List<Hearing> extractHearings(String payload, String messageId) {
     }
 
     private List<Hearing> parseCPMessage(SnsMessageContainer snsMessageContainer) throws JsonProcessingException {
-        final var cpHearingEvent = commonPlatformParser.parseMessage(snsMessageContainer.getMessage(), CPHearingEvent.class);
+        String message = snsMessageContainer.getMessage();
+
+        if (snsMessageContainer.getMessageAttributes().getExtendedPayloadSize() != null) {
+            message = getPayloadFromS3(snsMessageContainer);
+            log.info("Successfully downloaded large message from s3 bucket with extended payload size: {}", snsMessageContainer.getMessageAttributes().getExtendedPayloadSize());
+        }
+        final var cpHearingEvent = commonPlatformParser.parseMessage(message, CPHearingEvent.class);
         final var hearing = cpHearingEvent.asDomain();
         return passHearingIdToCourtCaseService ? setHearingAttributes(hearing, cpHearingEvent, snsMessageContainer) : hearing;
```

Comment:
I know you haven't touched this, but it is always true. You could perhaps take the opportunity to remove this condition and associated feature flag :-)